tests/trainer/loops/test_evaluation_loop_flow.py

```diff
@@ -65,6 +66,8 @@ def backward(self, loss, optimizer, optimizer_idx):
     assert not model.validation_step_end_called
     assert not model.validation_epoch_end_called
 
+    trainer.state.stage = RunningStage.TRAINING
```

Comment:
this is here to simulate the behavior we are still training right? so that you can make the assertions below?